### PR TITLE
RHINENG-8218 dependencies upgrade: react 18, patternfly V5 and more

### PR DESCRIPTION
**Pull request by PreetiW**
## React 18 & Patternfly V5 upgrade :boom:

RHINENG-8218 dependencies upgrade: react 18, patternfly V5 and more

**Jira:**

https://issues.redhat.com/browse/RHINENG-8218
https://issues.redhat.com/browse/RHINENG-8219

## Why do we need this change? :thought_balloon:

Please include the __context of this change__ here.

## Documentation requires update? :memo:

- [ ] Yes
- [ ] No

## Security Checklist :lock:

Upon raising this PR please go through [RedHatInsights/secure-coding-checklist](https://github.com/RedHatInsights/secure-coding-checklist)

## :guardsman: Checklist :dart:

- [ ] Bugfix
- [ ] New Feature
- [ ] Refactor
- [ ] Unittests Added
- [ ] DRY code
- [ ] Dependency Added

## Additional :mega:

Feel free to add any other relevant details such as __links, notes, screenshots__, here.

### CODE REVIEW

File: src/Components/SystemDetail/RecommendationsTable.js [hunk 1]
```diff
@@ -9,147 +17,149 @@ import { TextContent, Text, TextVariants } from '@patternfly/react-core';
...
-            >
-                <Thead>
+                <Tbody key={'Developer program 1'} isExpanded={this.state.expanded}>
                     <Tr>
-                        <Th>Instance</Th>
-                        <Th>Suggested instances that fit the load better</Th>
+                        <Td expand={{
+                            rowIndex: index,
+                            isExpanded: this.state.expanded,
+                            onToggle: () => this.onExpandToggle(),
```
Comment (LiorKGOW) [inline src/Components/SystemDetail/RecommendationsTable.js:52]: This line can be converted to `onToggle: this.onExpandToggle`
The onToggle property should get a function which will be evoked when the toggle event is triggered, therefore there is no need to create an arrow function that calls this.onExpandToggle function in this case
If I am not mistaken 😇

**Review by LiorKGOW — COMMENTED** [review]
Thank you for the PR @PreetiW ! 
I left a small comment, but will get back to reviewing this PR a bit later today 🙏🏼

**Comment by johnsonm325** [comment]
I'm going through the changes, line-by-line, but I wanted to bring up a couple of things:

1. Is there an account I can log into to test the system detail table? you can DM me
2. This may be a good time to convert your class components to functional components. It's up to you, and you can do it here or in another PR. I won't hold up approving this PR on that though. But, I will say that in other PRs where class components were used, I noticed strange behavior until I converted the components.

**Review by johnsonm325 — CHANGES_REQUESTED** [review]
Everything looks okay, but I want to check functionality with some systems. Currently, the qa account only has deleted systems so I can't test full functionality.

**Review by johnsonm325 — CHANGES_REQUESTED** [review]
On the system details page, the gap between the "Delete" button and the "Actions" dropdown is gone.
image

Also, if you look at the comparison of these two (your changes are on the left and stage is on the right) it appears that your changes have removed the blue highlight on the expanded which (which I know it should have).
image

File: src/Components/SystemDetail/RecommendationsTable.js (hunk 1 above)
Comment (PreetiW) [reply src/Components/SystemDetail/RecommendationsTable.js:52]: @LiorKGOW Thanks for picking this, I have updated the code in the latest commit. 🙌🏻

**Comment by PreetiW** [comment]
> I'm going through the changes, line-by-line, but I wanted to bring up a couple of things:
> 
> 1. Is there an account I can log into to test the system detail table? you can DM me
> 2. This may be a good time to convert your class components to functional components. It's up to you, and you can do it here or in another PR. I won't hold up approving this PR on that though. But, I will say that in other PRs where class components were used, I noticed strange behavior until I converted the components.

@johnsonm325 
1. I have shared the login cred in DM
2. Thank you for sharing, I didn't face any issues yet with code changes. There is plan to do refactoring as a part of separate card.

**Comment by PreetiW** [comment]
> @PreetiW I have noticed you removed tests for some components (like RecommendationRating.test.js) Are you planning on re-implementing them using React Testing Library ?
> 
> I am not talking about ones that only test snapshots, though it would be wise to implement a small unit test for each component to see that it's components are being rendered correctly, maybe check a small functionality like the expected modal is being opened

@LiorKGOW yes I have created a seperate card to add the tests using RTL and currently working on the same. I will raise seperate MR for it 🙌🏻  https://issues.redhat.com/browse/RHINENG-9113

**Comment by PreetiW** [comment]
> On the system details page, the gap between the "Delete" button and the "Actions" dropdown is gone. image
> 
> Also, if you look at the comparison of these two (your changes are on the left and stage is on the right) it appears that your changes have removed the blue highlight on the expanded which (which I know it should have). image

@johnsonm325 the delete and action button is handled by inventory component.

**Review by johnsonm325 — APPROVED** [review]
This looks good!

**Comment by johnsonm325** [comment]
@PreetiW You can bypass Lior's review here. He's passed off the review to me.

**Comment by PreetiW** [comment]
Thank you @johnsonm325 & @LiorKGOW for the review 🙌🏻 Merging this one!

**Pull request merged**